java-checks/src/main/java/org/sonar/java/checks/ServletInstanceFieldCheck.java

```diff
@@ -107,11 +109,19 @@ private void reportIssuesOnVariable() {
   private class AssignmentVisitor extends BaseTreeVisitor {
     @Override
     public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+      // handles e.g. "second = this.first * 2;" assignments -> no "this" prefix to member "second"
       if (tree.variable().is(Kind.IDENTIFIER)) {
-        Tree declaration = ((IdentifierTree) tree.variable()).symbol().declaration();
-        if (declaration != null && declaration.is(Kind.VARIABLE)) {
-          excludedVariables.add((VariableTree) declaration);
-        }
+        maybeAddExcluded(((IdentifierTree) tree.variable()).symbol().declaration());
+      }
+      // handles e.g. "this.first = 42;" assignments -> member "first" is prefixed with "this."
+      if (tree.variable().is(Kind.MEMBER_SELECT)) {
+        maybeAddExcluded(((MemberSelectExpressionTree) tree.variable()).identifier().symbol().declaration());
+      }
+    }
+    private void maybeAddExcluded(@Nullable Tree declaration) {
```

Comment:
Maybe something like `addExcludedIfVaraible` or `addVariableToExcluded` would be a more descriptive name here?